internal/pkg/term/selector/selector.go

```diff
@@ -945,27 +937,36 @@ func (s *LocalEnvironmentSelector) LocalEnvironment(msg, help string) (string, e
 }
 
 func filterEnvsByName(envs []*config.Environment, wantedNames []string) []string {
-	// TODO: refactor this and `filterWlsByName`  when generic supports using common struct fields: https://github.com/golang/go/issues/48522
-	isWanted := make(map[string]bool)
-	for _, name := range wantedNames {
-		isWanted[name] = true
-	}
-	var filtered []string
-	for _, wl := range envs {
-		if _, ok := isWanted[wl.Name]; !ok {
-			continue
-		}
-		filtered = append(filtered, wl.Name)
-	}
-	return filtered
+	return FilterItemsByStrings(wantedNames, envs, func(e *config.Environment) string { return e.Name })
 }
 
 func filterWlsByName(wls []*config.Workload, wantedNames []string) []string {
-	stringWls := make([]string, len(wls))
-	for i := range wls {
-		stringWls[i] = wls[i].Name
+	return FilterItemsByStrings(wantedNames, wls, func(w *config.Workload) string { return w.Name })
+}
+
+// FilterItemsByStrings is a generic function to return the subset of wantedStrings that exists in possibleItems.
+// stringFunc is a method to convert the generic item type (T) to a string; for example, one can convert a struct of type
+// *config.Workload to a string by passing
+//
+//	func(w *config.Workload) string { return w.Name }.
+//
+// Likewise, FilterItemsByStrings can work on a list of strings by returning the unmodified item:
+//
+//	FilterItemsByStrings(wantedStrings, stringSlice2, func(s string) string { return s })
+//
+// It returns a list of strings (items whose stringFunc() exists in the list of wantedStrings).
+func FilterItemsByStrings[T any](wantedStrings []string, possibleItems []T, stringFunc func(T) string) []string {
```

Comment:
My understanding is that we export this because `cli` needs the same logic, too. My only concern is that this function is really not specific to `selector` - I was quite confused when I saw `selector.FilterItemsByString` in the `cli` package, because I assumed some selection questions were being asked.

This function probably would have belong to a package named `util`. But we don't have that (because it is easy for `util` to just become dumping ground for all sort of random functions). 

Personally, I feel like it's okay to have some duplicate logic - the runtime performance isn't affected anyway, and the duplication isn't making the code harder to maintain 🤔 On the other hand, `FilterItemsByString` doesn't belong to `selector`. This looks like a trade-off between duplicating codes vs. having a weird package interface, and I'd like to vote for the former!